plugins/modules/redfish_firmware_rollback.py

```diff
@@ -0,0 +1,360 @@
+NO_CHANGES_FOUND = "No changes found to be applied."
+NOT_SUPPORTED = "The target firmware version does not support the firmware rollback."
+COMPLETED_ERROR = "The job for firmware rollback has been completed with error(s)."
+SCHEDULED_ERROR = "The job for firmware rollback has been scheduled with error(s)."
+ROLLBACK_SUCCESS = "Successfully completed the job for firmware rollback."
+ROLLBACK_SCHEDULED = "Successfully scheduled the job for firmware rollback."
+ROLLBACK_FAILED = "Failed to complete the job for firmware rollback."
+REBOOT_FAIL = "Failed to reboot the server."
+NEGATIVE_TIMEOUT_MESSAGE = "The parameter reboot_timeout value cannot be negative or zero."
+JOB_WAIT_MSG = "Task excited after waiting for {0} seconds. Check console for firmware rollback status."
+REBOOT_COMP = ["Integrated Dell Remote Access Controller"]
+
+
+def get_rollback_preview_target(redfish_obj, module):
+    action_resp = redfish_obj.invoke_request("GET", "{0}{1}".format(redfish_obj.root_uri, UPDATE_SERVICE))
+    action_attr = action_resp.json_data["Actions"]
+    update_uri = None
+    if "#UpdateService.SimpleUpdate" in action_attr:
+        update_service = action_attr.get("#UpdateService.SimpleUpdate")
+        if 'target' in update_service:
+            update_uri = update_service.get('target')
+        else:
+            module.fail_json(msg=NOT_SUPPORTED)
+    inventory_uri = action_resp.json_data.get('FirmwareInventory').get('@odata.id')
+    inventory_uri_resp = redfish_obj.invoke_request("GET", "{0}{1}".format(inventory_uri, "?$expand=*($levels=1)"),
+                                                    api_timeout=120)
+    previous_component = list(filter(lambda d: d["Id"].startswith("Previous"), inventory_uri_resp.json_data["Members"]))
+    if not previous_component:
+        module.fail_json(msg=NO_COMPONENTS)
+    component_name = module.params["name"]
+    try:
+        component_compile = re.compile(r"^{0}$".format(component_name))
+    except sre_constants.error:
+        component_compile = ""
+        module.exit_json(msg=NO_CHANGES_FOUND)
+    prev_uri, reboot_uri = {}, []
+    for each in previous_component:
+        available_comp = each["Name"]
+        available_name = re.match(component_compile, available_comp)
+        if not available_name:
+            continue
+        if available_name.group() in REBOOT_COMP:
+            reboot_uri.append(each["@odata.id"])
+            continue
+        prev_uri[each["Version"]] = each["@odata.id"]
+    if module.check_mode and (prev_uri or reboot_uri):
+        module.exit_json(msg=CHANGES_FOUND, changed=True)
+    elif (module.check_mode or not module.check_mode) and (not prev_uri and not reboot_uri):
+        module.exit_json(msg=NO_CHANGES_FOUND)
+    return list(prev_uri.values()), reboot_uri, update_uri
+
+
+def get_job_status(redfish_obj, module, job_ids, job_wait=True):
+    each_status, failed_count, js_job_msg = [], 0, ""
+    wait_timeout = module.params["reboot_timeout"]
```

Comment:
use .get otherwise will throw KeyError if reboot_timeout is not found.

Reply:
`reboot_timeout` - option has default 900 so not required to use `get`